exec/java-exec/src/main/java/org/apache/drill/exec/planner/common/DrillRelOptUtil.java

```diff
@@ -169,4 +176,223 @@ private static boolean containIdentity(List<? extends RexNode> exps,
     }
     return true;
   }
+
+  /**
+   * Copied from {@link RelOptUtil#splitJoinCondition(RelNode, RelNode, RexNode, List, List)}. Modified to rewrite
```

Comment:
Agree that we ideally should leverage the Calcite code..especially since this method is pretty heavily used and modified periodically so keeping Drill's version of this method in sync will be difficult.